security/pkg/pki/util/keycertbundle.go

```diff
@@ -224,6 +225,11 @@ func verify(certBytes, privKeyBytes, certChainBytes, rootCertBytes []byte) error
 			"cannot verify the cert with the provided root chain and cert pool")
 	}
 
+	// certificate must be usable as CA (including self signed)
+	if bytes.Compare(cert.RawIssuer, cert.RawSubject) != 0 && !cert.IsCA {
```

Comment:
Do you need to check "bytes.Compare(cert.RawIssuer, cert.RawSubject) != 0"? I think even for self-signed certs, they must have cert.IsCA==true to sign workload certs, right?

Reply:
I would expect so.
Many of the existing certificates used in tests are created without the correctKeyUsage. I will update with newly created certificates.

Reply:
Since the cert bundle is also used (in tests) to validate workload certificates, test test for `IsCA` will be moved to Cidatdel main code.